kiwi/builder/disk.py

```diff
@@ -287,225 +327,70 @@ def create_disk(self) -> Result:
         ) as loop_provider:
             loop_provider.create()
 
-            disk = Disk(
-                self.firmware.get_partition_table_type(), loop_provider,
-                self.disk_start_sector,
-                extended_layout=bool(self.dosparttable_extended_layout)
-            )
+            bootloader_custom_parameters = {
+                'targetbase':
+                    loop_provider.get_device(),
+                'grub_directory_name':
+                    Defaults.get_grub_boot_directory_name(
+                        self.root_dir
+                    ),
+                'crypto_disk':
+                    True if self.luks is not None else False,
```

Comment:
This should do the trick too
```suggestion
                'crypto_disk': bool(self.luks),
```

Reply:
nope, an empty passphrase for luks is allowed and still crypto

```python
luks = ''
print(luks)
x = True if luks is not None else False
print(x)
x = bool(luks)
print(x)
```

I stumbled over exactly that problem in the past ;)